examples/common/pigweed/rpc_services/BooleanState.h

```diff
@@ -28,7 +28,7 @@
 namespace chip {
 namespace rpc {
 
-class BooleanState final : public pw_rpc::nanopb::BooleanState::Service<BooleanState>
```

Comment:
Is this correct thoug? could these be actually meant to be `override` instead rather than virtual that is supposed to be overriden later?

If these are override, it does sound reasonable for these classes to be final.

Reply:
I did't get this, a final class cannot be inherited from, typically does not need member functions with the virtual keyword. The virtual keyword is used to allow derived classes to override functions in the base class. Since a final class cannot be inherited, there will be no derived classes to override its member functions.

After looking at other interface definitions, the interface definitions within these header files might be override by different platforms, such as https://github.com/project-chip/connectedhomeip/blob/master/examples/platform/esp32/Rpc.cpp#L148.

If the member functions are not meant to be overridden, they should not be marked with the virtual keyword. Conversely, if the functions are marked with virtual, the class should not be declared as final.

Reply:
My question is: in what situations will we override `BooleanState` ?

Reply:
Ok, I thought that these were implementations overriding the BooleanState::Service ... looking at the impl that does not seem to be what is happening here.